cypress/e2e/alerts.cy.js

```diff
@@ -1,25 +1,57 @@
+/// <reference types='cypress' />
+const faker = require('faker');
 describe('Cypress application', () => {
-  before(() => {
-
+  beforeEach(() => {
+    cy.visit('/');
   });
 
-  it('should have the ability to assert automatically resolved alerts', () => {
+  it('should be able to assert alerts that are resolved automatically', () => {
+    cy.get('#alertButton').click();
 
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.equal('You clicked a button');
+    });
   });
 
-  it('should have the ability to assert scheduled allert', () => {
+  it('should be able to assert alerts that are scheduled', () => {
+    cy.get('#timerAlertButton').click();
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(5000);
 
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.equal('This alert appeared after 5 seconds');
+    });
   });
 
-  it('should autimatically resolve alerts', () => {
-
+  it('should randomly resolve or cancel alerts', () => {
+    const randomChoice = Math.random() < 0.5;
+
+    cy.get('#confirmButton').click();
+
+    cy.on('window:confirm', (confirmText) => {
+      expect(confirmText).to.equal('Do you confirm action?');
+      return randomChoice;
+    });
+
+    if (randomChoice) {
+      cy.contains('#confirmResult', 'You selected Ok').then(() => {
+        cy.log('User chose: Ok');
+      });
+    } else {
+      cy.contains('#confirmResult', 'You selected Cancel').then(() => {
+        cy.log('User chose: Cancel');
+      });
+    }
```

Comment:
This conditional block is outside of the Cypress commands, which means it will be executed immediately, not in the order of the commands. This can lead to unexpected behavior. It's recommended to use `.then()` to ensure commands are executed in the correct order. For example: `cy.wrap(randomChoice).then((choice) => { if (choice) {...} else {...} })`.